packages/mobile/src/utils/navigation.ts

```diff
@@ -21,3 +21,19 @@ export const getPrimaryRoute = (state: NavigationState) => {
   // The route at index 2 is the primary route
   return getRoutePath(state)?.[2]
 }
+
+/**
+ * Given a navigator, get the nearest stack navigator in the hierarchy
+ */
+export const getNearestStackNavigator = (navigator: NavigationProp<any>) => {
+  if (navigator.getState?.()?.type === 'stack') {
+    return navigator
+  }
+  const parent = navigator.getParent()
+
+  if (!parent) {
+    return undefined
```

Comment:
should we return `navigator` in this case?

Reply:
I think it's safer to have the consumer explicitly check that there is a result, let me fix up the types so that consumers are forced to check